assignments/sidharth/week1/1.7.scm

```diff
@@ -0,0 +1,15 @@
+#|
+The problem with finding square root of very small numbers using this method is that The logic prevents the improvement of the guess past the 0.001 hardcoded threshold since the radicant itself is much smaller than this threshold.
+
+The problem with finding square root of very large numbers is that the code never stops running. This is because floats have a fixed precision. After a certain point, the guesses are unable to guess the square root of a number within 0.001 of the answer. At that point, they alternate between the two guesses, both of whose squares are more than 0.001 away from the argument.
+|#
+
+(define (average x y) (/ (+ x y) 2))
+(define (improve guess x) (average guess (/ x guess)))
+(define (goodenough? guess prev-guess)
+  (<= (abs (- prev-guess guess)) (/ guess 100)))
```

Comment:
The evaluation assumes the goodenough? test uses (/ guess 1000), or (* guess 0.001). e.g. the hardcoded value is the same magnitude as the previous implementation.